source/libs/wal/src/walMgmt.c

```diff
@@ -98,15 +104,14 @@ SWal *walOpen(const char *path, SWalCfg *pCfg) {
   tstrncpy(pWal->path, path, sizeof(pWal->path));
   if (taosMkDir(pWal->path) != 0) {
     wError("vgId:%d, path:%s, failed to create directory since %s", pWal->cfg.vgId, pWal->path, strerror(errno));
-    taosMemoryFree(pWal);
-    return NULL;
+    goto _err;
```

Comment:
There is no evil to use goto _err for handling resource cleanup. In the C programming which is lacking of RAII, it is most neat way to deal with resource acquisition/cleanup with it when used properly.